crates/evm/src/evm/db.rs

```diff
@@ -88,7 +88,7 @@ impl<'a, C: sov_modules_api::Context> Database for EvmDb<'a, C> {
         // TODO move to new_raw_with_hash for better performance
         Ok(self
             .code
-            .get(&code_hash, self.working_set)
+            .get(&code_hash, &mut self.working_set.offchain_state())
```

Comment:
once more this should be like this:

```rust
let code = self
            .code
            .get(&code_hash, &mut self.working_set.offchain_state()

assert_eq!(keccak256(code), code_hash);

Ok(code)
```

or you can return Err as well.

There is nothing preventing us from supplying the wrong code to the offchain witness right now (for zk proving)